osf/models/preprintlog.py

```diff
@@ -38,6 +38,7 @@ class PreprintLog(ObjectIDMixin, BaseModel):
     CHANGED_LICENSE = 'license_changed'
 
     FILE_UPDATED = 'file_updated'
+    FILE_REMOVED = 'file_removed'
```

Comment:
Why are these being changed if they aren't really being used (there's no `.add_log` call ect.) 

I know this is kinda bad practice but since this issue has been dogging us a while it might be best to just add a check constraint to posrtgres that will ensure the file is never null. While [check constraints aren't super well supported until Django 3](https://docs.djangoproject.com/en/3.0/ref/models/constraints/#checkconstraint) we can them now and it will just cancel any DB transaction that tries to remove the file.